src/components/NewFileDialog.tsx

```diff
@@ -46,33 +46,42 @@ export class NewFileDialog extends React.Component<{
       name: ""
     };
   }
+
   onChangeName = (event: ChangeEvent<any>) => {
     this.setState({ name: event.target.value });
   }
+
   getNameError() {
-    const directory = this.props.directory;
     if (this.state.name) {
-      if (!/^[a-z0-9\.\-\_]+$/i.test(this.state.name)) {
-        return "Illegal characters in file name.";
-      } else if (!this.state.name.endsWith(extensionForFileType(this.state.fileType))) {
-        return nameForFileType(this.state.fileType) + " file extension is missing.";
-      } else if (directory && appStore.getImmediateChild(directory, this.state.name)) {
-        return `File '${this.state.name}' already exists.`;
+      const fileNameError: string = validateFileName(this.state.name, this.state.fileType);
+      if (fileNameError) {
+        return fileNameError;
+      }
+
+      const directory = this.props.directory;
+      if (directory && appStore.getImmediateChild(directory, this.state.name)) {
+        return `File '${this.state.name}' already exists`;
       }
     }
+
```

Comment:
nit: remove unrelated changed such as empty lines or indentation -- it is expected to have separate PR for that.